Refactor post and comment handling; enhance sharing logic

Pdzly opened the pull request:
Refactored post and comment functionalities by introducing `MemePost` and optimizing metadata handling. Updated comment scroll behavior and added an advanced sharing flow via Web Share API and Clipboard.

I did NOT implement the case when there are too many comments to fit in one batch ( and "when" replies are implemented that needs to handled too )

![grafik](https://github.com/user-attachments/assets/ab1e086e-26cc-41c2-9b2f-a113e2096c82)

URL: 
http://local-whysodank.com/posts/a8316130c26f43ff8639006dc670c86c/comment/5249bbecbf0b4c32acc68d21d3f7330a

Title will be:
`user.username - comment on post.title`

One thing i could not find out how to do really is for metadata the html body text.

Pdzly committed:
Refactor post and comment handling; enhance sharing logic

Refactored post and comment functionalities by introducing `MemePost` and optimizing metadata handling. Updated comment scroll behavior and added an advanced sharing flow via Web Share API and Clipboard. Adjusted configurations and documentation for frontend and container compatibility improvements.

isik-kaplan commented:
Can you also split this into two PRs so we can work on the dev environment part in another PR?

I think instead of having complete different compose files maybe we can use `extends` feature of docker compose which may allow us to just change small parts instead of repeating everything - also instead of a docker-dev-frontend docker-dev-backend etc I'd prefer a single docker-compose-dev that has volumes to both front-end and back-end code and works with auto refreshes on both backend and front-end

Pdzly commented:
I tried to do that always, but the auto refresh is really unreliable "though" docker. But we can still try it.

Pdzly committed:
Reverted Local Frontend

Pdzly committed:
Remove console.log statements from MemeComment component

Removed unnecessary console.log statements used for debugging postId and commentId. This cleanup improves code readability and avoids cluttering the console during runtime.

Pdzly committed:
Update local-development.md

Pdzly committed:
Update example.env

Pdzly committed:
Update docker-compose.yml

Pdzly committed:
Update docker-compose.yml

Pdzly committed:
Update monkeypatches.ts

Pdzly committed:
Merge remote-tracking branch 'upstream/master' into feature/comment_link

# Conflicts:
#	wsd-frontend/src/app/(app)/posts/[hex]/page.tsx

isik-kaplan commented on wsd-frontend/src/components/wsd/MemePost/MemePost.tsx:
```diff
@@ -0,0 +1,155 @@
+          if (!_.isUndefined(comment)) {
+            const commentData = includesType(comment, 'user', 'User')
+
+            // @TODO: Handle comment body
```
Are you looking to render the comment and then return the text? That might be slightly complicated, we can render the comments on the backend but we can't get the text easily, we can throw in something like jsdom to emulate dom and use some dom method to get the text, that's the only method I can think of right now.

Pdzly replied:
Yes but i dont know how to do it really, i would rather butcher your fine code even more.

isik-kaplan commented on wsd-frontend/src/components/wsd/MemeComment/MemeComment.tsx:
```diff
@@ -114,6 +167,14 @@ export function MemeComment({ comment }: { comment: Includes<APIType<'PostCommen
+            <Button
+              onClick={handleShare}
+              size="sm"
+              className="flex items-center gap-1 p-2 rounded-md transition-colors text-gray-500 hover:bg-secondary bg-transparent"
```
Can we keep the colors consistent with other buttons? Also design wise do you think we should remove the hover background like upvote and downvote buttons?

isik-kaplan commented on wsd-frontend/src/components/wsd/MemeComment/MemeComment.tsx:
```diff
@@ -114,6 +167,14 @@ export function MemeComment({ comment }: { comment: Includes<APIType<'PostCommen
+              className="flex items-center gap-1 p-2 rounded-md transition-colors text-gray-500 hover:bg-secondary bg-transparent"
+              aria-label="Share"
+            >
+              <Icons.LucideLink size={20} />
```
Can you try 16 instead of 20, that looks more natural, somehow the link icon is bigger than others (maybe has some padding baked in to the svg) Also another nit, can you change it to Icons.Link instead of Icons.LucideLink

isik-kaplan reviewed (commented):
I think we can merge this if you just move some code around and optionally add the comment body to the metadata (if you want I can handle that part) 

My comments are mostly ocd fueled nitpicks and trying to follow some unwritten conventions I have: 
- Keep the metadata functions definitions in the page component files
- 1 component 1 file (for stuff under /wsd/) 
- Write reusable stuff if we can and prioritize declarative ui instead of imperative 
- Try to keep the ui consistent as much as we can even if the previous decisions were wrong(if there are wrong we should try to change it all at once so that the ui is still consistent) 

But other than these this is a great addition thank you. 

I also have a design question:

![image](https://github.com/user-attachments/assets/729b1008-8d20-40eb-84e3-23d9eb6a2914)

I made the background black for the selected comment, do you think this makes it "pop" more? Totally your decision, if you want we can keep it with the border/outline you have

Pdzly commented on wsd-frontend/next.config.mjs:
```diff
@@ -1,73 +1,75 @@
+    authInterrupts: true,
+    optimizePackageImports: ['lodash'],
+  },
+  allowedDevOrigins: ['localhost:3000', 'local-whysodank.com'],
```
yeah will remove it. ( I get always a warning that you should add dev origins to that etc... )

Pdzly committed:
Reverted next.config.mjs

Pdzly committed:
Remove leftover debug log from comment page component

Removed an unnecessary console.log statement that appeared to be a debugging artifact. This change cleans up the code and improves maintainability by removing unused code.

Pdzly committed:
Moved Metadata logik back to the page.tsx

Pdzly commented:
![grafik](https://github.com/user-attachments/assets/ef3ba687-e08c-4462-a024-7da4a78deb6b)
Updated style

Pdzly committed:
Refactor scroll logic with ScrollToHashContainer in comments

Replaced custom scroll logic in `MemeComment` with the reusable `ScrollToHashContainer` component. Improved code structure by simplifying properties while preserving existing functionality for smooth scrolling and highlighting targeted comments.

Pdzly commented:
@isik-kaplan done all of your comments, let me know if this is ok. 

The only thing that is left is the comment body parsing.

isik-kaplan commented:
This looks good, thank you, I'll add the comment body stuff asap and merge it

Pdzly committed:
Merge remote-tracking branch 'refs/remotes/upstream/master' into feature/comment_link

# Conflicts:
#	wsd-frontend/next.config.mjs
#	wsd-frontend/src/app/(app)/posts/[hex]/page.tsx
#	wsd-frontend/src/components/wsd/MemeComment/MemeComment.tsx

Pdzly committed:
Refactor WSD API hook usage and enhance `includesType` call in MemePost component
